public/tier1/convar.h

```diff
@@ -395,64 +272,37 @@ inline const char *CCommand::operator[]( int nIndex ) const
 //-----------------------------------------------------------------------------
 // Purpose: The console invoked command
 //-----------------------------------------------------------------------------
-class ConCommand : public ConCommandBase
+struct ConCommandCreation_t : CVarCreationBase_t
 {
-friend class CCvar;
-friend class ConCommandHandle;
-
-public:
-	typedef ConCommandBase BaseClass;
-
-	ConCommand( ConCommandRefAbstract *pReferenceOut, const char *pName, FnCommandCallback_t callback,
-		const char *pHelpString = 0, int64 flags = 0, FnCommandCompletionCallback completionFunc = 0 );
-	ConCommand( ConCommandRefAbstract *pReferenceOut, const char *pName, FnCommandCallbackVoid_t callback,
-		const char *pHelpString = 0, int64 flags = 0, FnCommandCompletionCallback completionFunc = 0 );
-	ConCommand( ConCommandRefAbstract* pReferenceOut, const char* pName, FnCommandCallbackNoContext_t callback,
-		const char* pHelpString = 0, int64 flags = 0, FnCommandCompletionCallback completionFunc = 0 );
-	ConCommand( ConCommandRefAbstract *pReferenceOut, const char *pName, ICommandCallback *pCallback,
-		const char *pHelpString = 0, int64 flags = 0, ICommandCompletionCallback *pCommandCompletionCallback = 0 );
-
-	~ConCommand( void );
-
-	// Used internally by OneTimeInit to initialize/shutdown
-	void Init();
-	void Shutdown();
+	ConCommandCreation_t() :
+	m_bHasCompletionCallback(false),
+	m_bIsInterface(false),
+	m_pHandle(nullptr)
+	{}
 
-	void Create( const char *pName, const char *pHelpString = 0,
-		int64 flags = 0 );
-
-	int AutoCompleteSuggest( const char *partial, CUtlVector< CUtlString > &commands );
-
-	bool CanAutoComplete( void );
-
-	inline ConCommandRefAbstract	*GetRef( void ) const
-	{
-		return m_pReference;
-	}
-
-	inline void SetHandle( ConCommandHandle hndl )
-	{
-		m_pReference->handle = hndl;
-	}
-
-private:
 	// Call this function when executing the command
-	class CallbackInfo_t
+	struct CallbackInfo_t
 	{
-	public:
+		CallbackInfo_t() :
+		m_fnCommandCallback(nullptr),
+		m_bIsInterface(false),
+		m_bIsVoidCallback(false),
+		m_bIsContextLess(false)
+		{}
+
 		union {
 			FnCommandCallback_t m_fnCommandCallback;
 			FnCommandCallbackVoid_t m_fnVoidCommandCallback;
 			FnCommandCallbackNoContext_t m_fnContextlessCommandCallback;
 			ICommandCallback* m_pCommandCallback;
 		};
 
-		bool m_bUsingCommandCallbackInterface : 1;
-		bool m_bHasVoidCommandCallback : 1;
-		bool m_bHasContextlessCommandCallback : 1;
+		bool m_bIsInterface;
+		bool m_bIsVoidCallback;
+		bool m_bIsContextLess;
 	};
 
-	CallbackInfo_t m_Callback;
+	CallbackInfo_t m_fnCallback;
 
 	// NOTE: To maintain backward compat, we have to be very careful:

```

Comment:
How up to date this comment is? If this is a remnant of s1 stuff, I guess this could be removed